pkg/virt-handler/vm.go

```diff
@@ -235,6 +243,50 @@ func (d *VirtualMachineController) hasTargetDetectedDomain(vmi *v1.VirtualMachin
 	return false, timeLeft
 }
 
+func (d *VirtualMachineController) clearPodNetworkPhase1(vmi *v1.VirtualMachineInstance) {
+	d.phase1NetworkSetupCacheLock.Lock()
```

Comment:
Why is this lock required?

Reply:
the controller's reconcile loop has multiple goroutines active for processing the work queue. The map I used here isn't threadsafe, so if we're going to do concurrent reads/writes, we need a lock to synchronize that.

Reply:
thanks